BUILD.bazel

```diff
@@ -0,0 +1,523 @@
+    srcs = [
+        "tools/cfg/bin_to_dot.cpp",
+        "tools/cfg/bin_to_dot.h",
+        "tools/cfg/cfg.cpp",
+        "tools/io.h",
+    ],
+    copts = COMMON_COPTS,
+    visibility = ["//visibility:public"],
+    deps = [":spirv_tools"],
+)
+
+# Unit tests
+
+cc_library(
+    name = "test_common",
+    testonly = 1,
+    srcs = [
+        "test/test_fixture.h",
+        "test/unit_spirv.cpp",
+        "test/unit_spirv.h",
+    ],
+    compatible_with = [],
+    copts = TEST_COPTS,
+    includes = ["test"],
+    linkstatic = 1,
+    deps = [
+        ":spirv_tools",
+        "@com_google_googletest//:gtest",
+    ],
+)
+
+cc_library(
+    name = "link_test_common",
+    testonly = 1,
+    srcs = ["test/link/linker_fixture.h"],
+    compatible_with = [],
+    copts = TEST_COPTS,
+    linkstatic = 1,
+    deps = [
+        ":spirv_tools_link",
+        ":test_common",
+    ],
+)
+
+cc_library(
+    name = "opt_test_common",
+    testonly = 1,
+    srcs = ["test/opt/pass_utils.cpp"],
+    hdrs = [
+        "test/opt/assembly_builder.h",
+        "test/opt/function_utils.h",
+        "test/opt/module_utils.h",
+        "test/opt/pass_fixture.h",
+        "test/opt/pass_utils.h",
+    ],
+    compatible_with = [],
+    copts = TEST_COPTS,
+    linkstatic = 1,
+    deps = [
+        ":spirv_tools_opt",
+        ":test_common",
+    ],
+)
+
+cc_library(
+    name = "reduce_test_common",
+    testonly = 1,
+    srcs = [
+        "test/reduce/reduce_test_util.cpp",
+        "tools/io.h",
+    ],
+    hdrs = ["test/reduce/reduce_test_util.h"],
+    compatible_with = [],
+    copts = TEST_COPTS,
+    linkstatic = 1,
+    deps = [
+        ":spirv_tools_reduce",
+        ":test_common",
+    ],
+)
+
+cc_library(
+    name = "val_test_common",
+    testonly = 1,
+    srcs = [
+        "test/val/val_code_generator.cpp",
+        "test/val/val_fixtures.h",
+    ],
+    hdrs = [
+        "test/val/val_code_generator.h",
+    ],
+    compatible_with = [],
+    copts = TEST_COPTS,
+    linkstatic = 1,
+    deps = [":test_common"],
+)
+
+[base_test(
+    name = f[5:-9],  # strip test/, _test.cpp
+    srcs = [f],
+) for f in glob(
+    ["test/*_test.cpp"],
+    exclude = [
+        "test/cpp_interface_test.cpp",
+        "test/log_test.cpp",
+        "test/pch_test.cpp",
+        "test/timer_test.cpp",
+    ],
+)]
+
+base_test(
+    name = "cpp_interface_test",
+    srcs = ["test/cpp_interface_test.cpp"],
+    deps = [":spirv_tools_opt"],
+)
+
+base_test(
+    name = "ilist_test",
+    srcs = ["test/util/ilist_test.cpp"],
+)
+
+base_test(
+    name = "log_test",
+    srcs = ["test/log_test.cpp"],
+    deps = [":spirv_tools_opt"],
+)
+
+[link_test(
+    name = f[10:-9],  # strip test/link/, _test.cpp
+    srcs = [f],
+) for f in glob(
+    ["test/link/*_test.cpp"],
+    exclude = ["test/link/type_match_test.cpp"],
+)]
+
+[opt_test(
+    name = f[9:-9],  # strip test/opt/, _test.cpp
+    srcs = [f],
+) for f in glob(["test/opt/*_test.cpp"])]
```

Comment:
This will miss some tests like `ir_builder.cpp`.  It forces a particular naming convention that we general keep, but we have not enforced.  We should try to make it more general.

Reply:
The same goes for other places we look for `*_test.cpp`.

Reply:
Done